src/Glpi/Database.php

```diff
@@ -0,0 +1,1215 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2018 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi;
+
+use \PDO;
+use \DBmysqlIterator;
+use \Session;
+use \Timer;
+use \Toolbox;
+
+if (!defined('GLPI_ROOT')) {
+    die("Sorry. You can't access this file directly");
+}
+
+abstract class Database
+{
+    //! Database Host - string or Array of string (round robin)
+    protected $dbhost;
+    //! Database User
+    protected $dbuser;
+    //! Database Password
+    protected $dbpassword;
+    //! Default Database
+    protected $dbdefault;
+    //! Database Handler
```

Comment:
Documenting properties type will be usefull for IDE and code analysis tools.
```suggestion
    /**
     * Database handler.
     * @var PDO
     */
```